src/index.ts

```diff
@@ -199,6 +208,46 @@ const LICENSE_ENV_VARIABLES = [
   "ACCEPT_MAX_GAS_PRICE_GWEI_VALUE",
 ];
 
+async function pollPooledMinerResults() {
+  const provider = getProvider();
+  const wallet = new Wallet(process.env.PRIVATE_KEY, provider);
+  const submittedNonces = [];
+  while (true) {
+    let result = await axios.get<PooledMinerResult>(process.env.POOLED_MINER_RESULT_URL);
+    let last = result.data;
+    console.log(`polling pooled miner results for address ${wallet.address}, last mined address is ${last.address}...`)
+    if (last.address === wallet.address &&
+      last.error === null &&
+      last.success === true &&
+      submittedNonces.indexOf(last.nonce) == -1) {
+      console.log('found a valid nonce, will mint it if gas price is low enough');
+      const nonce = BigNumber.from(last.nonce);
```

Comment:
Just to be safe, I think we should call `checkNonce` here as well and confirm that the returned `nonceStatus` is `VALID` (similar to the logic in the `submit-work` endpoint)

Reply:
Donee